cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -582,6 +582,16 @@ public RunRecordDetail getRun(ProgramReference programRef, String runId) {
     });
   }
 
+  @Override
+  public void markApplicationsLatest(List<ApplicationId> appIds) throws IOException, ConflictException {
+    TransactionRunners.run(transactionRunner, context -> {
+      for (ApplicationId appId: appIds) {
+        ApplicationMeta appMeta = getApplicationMetadata(appId);
```

Comment:
I would expect this to throw an exception -- you can't start a transaction within another transaction.

It's also a bunch of unnecessary reads. We should just be updating a row for each appId given.

Reply:
The reads are necessary, as we also need to unmark the current latest version.

Reply:
The issue "start a transaction within another transaction" was because of using an incorrect method. Fixed that.